core/templates/dev/head/services/UserService.js

```diff
@@ -17,48 +17,55 @@
  */
 
 oppia.factory('UserService', [
-  '$http', '$q', '$window', 'UrlInterpolationService', 'UserInfoObjectFactory',
-  'DEFAULT_PROFILE_IMAGE_PATH',
-  function($http, $q, $window, UrlInterpolationService, UserInfoObjectFactory,
-      DEFAULT_PROFILE_IMAGE_PATH) {
+  '$http', '$q', '$window', 'UrlInterpolationService',
+  'UserInfoObjectFactory', 'DEFAULT_PROFILE_IMAGE_PATH',
+  function($http, $q, $window, UrlInterpolationService,
+      UserInfoObjectFactory, DEFAULT_PROFILE_IMAGE_PATH) {
     var PREFERENCES_DATA_URL = '/preferenceshandler/data';
 
     var userInfo = null;
 
     var getUserInfoAsync = function() {
-      if (GLOBALS.userIsLoggedIn) {
-        if (userInfo) {
-          return $q.resolve(userInfo);
-        }
-        return $http.get(
-          '/userinfohandler'
-        ).then(function(response) {
-          userInfo = UserInfoObjectFactory.createFromBackendDict(response.data);
-          return userInfo;
-        });
-      } else {
+      var currentUrl = $window.location.href;
+      if (currentUrl.includes('signup')) {
         return $q.resolve(UserInfoObjectFactory.createDefault());
       }
+      if (userInfo) {
+        return $q.resolve(userInfo);
+      }
+      return $http.get(
+        '/userinfohandler'
+      ).then(function(response) {
+        if (response.data.user_is_logged_in) {
+          userInfo =
+            UserInfoObjectFactory.createFromBackendDict(response.data);
+          return $q.resolve(userInfo);
+        } else {
+          return $q.resolve(UserInfoObjectFactory.createDefault());
```

Comment:
set `userInfo` to `UserInfoObjectFactory.createDefault()` here

Reply:
Done!